postgresw/src/main/java/com/arcadedb/postgres/PostgresNetworkExecutor.java

```diff
@@ -347,34 +347,37 @@ private void queryCommand() {
         return;
       }
 
-      if (DEBUG)
-        LogManager.instance().log(this, Level.INFO, "PSQL: query -> %s (thread=%s)", queryText, Thread.currentThread().getId());
+//      if (DEBUG)
+//        LogManager.instance().log(this, Level.INFO, "PSQL: query -> %s (thread=%s)", queryText, Thread.currentThread().getId());
 
       final Query query = getLanguageAndQuery(queryText);
+      if (DEBUG)
+        LogManager.instance().log(this, Level.INFO, "PSQL: query -> %s ", query);
 
       final ResultSet resultSet;
-      if (queryText.startsWith("SET ")) {
-        setConfiguration(queryText);
+      if (query.query.startsWith("SET ")) {
+        setConfiguration(query.query);
         resultSet = new IteratorResultSet(createResultSet("STATUS", "Setting ignored").iterator());
-      } else if (queryText.equals("SELECT VERSION()"))
+      } else if (query.query.equals("SELECT VERSION()"))
         resultSet = new IteratorResultSet(createResultSet("VERSION", "11.0.0").iterator());
-      else if (queryText.equals("SELECT CURRENT_SCHEMA()"))
+      else if (query.query.equals("SELECT CURRENT_SCHEMA()"))
         resultSet = new IteratorResultSet(createResultSet("CURRENT_SCHEMA", database.getName()).iterator());
-      else if (queryText.equalsIgnoreCase("BEGIN") || queryText.equalsIgnoreCase("BEGIN TRANSACTION")) {
+      else if (query.query.equalsIgnoreCase("BEGIN") ||
+          query.query.equalsIgnoreCase("BEGIN TRANSACTION")) {
         explicitTransactionStarted = true;
         database.begin();
         resultSet = new IteratorResultSet(Collections.emptyIterator());
-      } else if (ignoreQueries.contains(queryText))
+      } else if (ignoreQueries.contains(query.query))
         resultSet = new IteratorResultSet(Collections.emptyIterator());
       else
-        resultSet = database.command(query.language, query.query);
+        resultSet = database.query(query.language, query.query);
```

Comment:
Is changing `command` to `query` correct here? So changes to the database are handled in `parseCommand` below and `queryCommand` is readonly?

Reply:
Even if the problems are related, I'm splitting the fix in two different issues/prs